src/main/java/org/axion/axion_api_gateway/UserSession.java

```diff
@@ -67,7 +68,7 @@ private UserDTO fetchUserFromAuthService(String token) {
             }
             log.error("Unexpected response from auth service: {}. Message: {}", response.code(), response.message());
             throw new IllegalStateException("Unexpected response code: " + response.code());
-        } catch (Exception e) {
+        } catch (IOException e) {
```

Comment:
This generic Exception caused invalid 401 status handling when user session is not active. Should be specific to IOException check only for handling connection issues between microservices.